### PR TITLE
Added QuRT drivers for mpu9x50 and uart_esc

### PR DESCRIPTION
**Pull request by mcharleb**
Signed-off-by: Mark Charlebois <charlebm@gmail.com>

### CODE REVIEW

File: src/platforms/qurt/drivers/mpu9x50/module.mk [hunk 1]
```diff
@@ -0,0 +1,48 @@
+############################################################################
+#
+#   Copyright (c) 2012, 2013 PX4 Development Team. All rights reserved.
```
Comment (bansiesta) [inline src/platforms/qurt/drivers/mpu9x50/module.mk]: Adjust copyright.

File: src/platforms/qurt/drivers/uart_esc/module.mk [hunk 2]
```diff
@@ -0,0 +1,46 @@
+############################################################################
+#
+#   Copyright (c) 2012, 2013 PX4 Development Team. All rights reserved.
```
Comment (bansiesta) [inline src/platforms/qurt/drivers/uart_esc/module.mk]: Again, copyright.

File: src/platforms/qurt/drivers/uart_esc/uart_esc_main.cpp [hunk 3]
```diff
@@ -0,0 +1,465 @@
...
+void start(const char* device)
+{
+	ASSERT(_task_handle == -1);
+
+	_device = device;
+
+	/* start the task */
+	_task_handle = px4_task_spawn_cmd("uart_esc_main",
+					SCHED_DEFAULT,
+					SCHED_PRIORITY_MAX,
```
Comment (bansiesta) [inline src/platforms/qurt/drivers/uart_esc/uart_esc_main.cpp]: Does this task really deserve max priority?

File: src/platforms/qurt/drivers/uart_esc/uart_esc_main.cpp [hunk 4]
```diff
@@ -0,0 +1,465 @@
...
+	 */
+	if (!strcmp(verb, "start")) {
+		if (uart_esc::_is_running) {
+			PX4_WARN("uart_esc already running");
+			return 1;
+		}
+		uart_esc::start(device);
+	}
+
+  else if (!strcmp(verb, "stop")) {
```
Comment (bansiesta) [inline src/platforms/qurt/drivers/uart_esc/uart_esc_main.cpp]: indent...

File: src/platforms/qurt/drivers/uart_esc/uart_esc_main.cpp [hunk 5]
```diff
@@ -0,0 +1,465 @@
...
+				orb_copy(ORB_ID(parameter_update), _param_sub, &_param_update);
+				// The param update struct only contains a timestamp. You can use or not.
+				// Update parameters relevant to this task (TBD)
+			}
+		}
+	}
+
+	PX4_WARN("closing uart_esc");
+
+	delete esc;
```
Comment (bansiesta) [inline src/platforms/qurt/drivers/uart_esc/uart_esc_main.cpp]: esc can be nullptr here. I think you should move it up one bracket in.

File: src/platforms/qurt/drivers/uart_esc/uart_esc_main.cpp [hunk 6]
```diff
@@ -0,0 +1,465 @@
...
+	// Check on required arguments
+	if (device == NULL)
+	{
+		uart_esc::usage();
+		return 1;
+	}
+
+	const char *verb = argv[1];
+	PX4_WARN("verb = %s", verb);
+	PX4_WARN("result = %d", strcmp(verb, "start"));
```
Comment (bansiesta) [inline src/platforms/qurt/drivers/uart_esc/uart_esc_main.cpp]: are these debug printfs still needed?

**Comment by bansiesta** [comment]
I made a few small comments, other than that, it looks ok.

**Comment by LorenzMeier** [comment]
@bansiesta Thanks for the review, much appreciated! Would be awesome to have more of this. @mcharleb Could you address these comments?

**Comment by mcharleb** [comment]
The last two patches should address the comments above (much appreciated!). The 3rd last patch fixed build issues for QuRT.

File: src/platforms/qurt/drivers/uart_esc/uart_esc_main.cpp (hunk 5 above)
Comment (mcharleb) [inline src/platforms/qurt/drivers/uart_esc/uart_esc_main.cpp]: Fixed in the last patch below.

**Comment by mcharleb** [comment]
This pull request has been replaced with a new pull request that squashes all these changes into one commit.

**Pull request closed without merging**